### PR TITLE
fix : share still highlighted after cancellation

### LINKED ISSUE #1004: Share highlighted in NavDrawer after cancelling the share option
**Summary:**
Share highlighted in NavDrawer after cancelling the share option
**Steps to reproduce:**
Open Nav Drawer and select share
Cancel when you get options
**Expected behaviour:**
Home should be highlighted
**Observed behaviour:**
Share is highlighted

### PR DESCRIPTION
**Pull request by shiv07tiwari**
Fixes #1004 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by shiv07tiwari** [comment]
@luckyman20 @miPlodder

**Comment by luckyman20** [comment]
Fix your PR heading

**Comment by shiv07tiwari** [comment]
@luckyman20 fixed it.

**Review by luckyman20 — APPROVED** [review]
Looks good! Great job @shiv07tiwari

**Comment by shiv07tiwari** [comment]
@miPlodder @therajanmaurya Kindly review this PR.

File: app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java [hunk 1]
```diff
@@ -139,6 +139,7 @@ protected void onPause() {
     @Override
     protected void onResume() {
         super.onResume();
+        setNavigationViewSelectedItem(R.id.item_home);
```
Comment (miPlodder) [inline app/src/main/java/org/mifos/mobilebanking/ui/activities/HomeActivity.java]: @shiv07tiwari The changes you have made are incorrect. You cannot always change the selected Navigation Item to `Home` on `onResume()` method. Checkout Activity lifecycle, `onResume()` method is called many times.

A simple scenario whereas you solution will fail will be Go to `Charges > Open Another App > Now open Self-Service App > Charges Fragment` will be displayed and Home Navigation will be selected.

**Review by miPlodder — CHANGES_REQUESTED** [review]

**Comment by shiv07tiwari** [comment]
@miPlodder @luckyman20 Kindly review this. I added a check for item initially highlighted which solves the issue.

**Comment by shiv07tiwari** [comment]
@miPlodder @therajanmaurya Kindly review this.

**Comment by shiv07tiwari** [comment]
@miPlodder @therajanmaurya Kindly review this.

**Comment by shiv07tiwari** [comment]
@therajanmaurya Fixed the merge conflict.

**Comment by prashantkh19** [comment]
@shiv07tiwari I couldn't reproduce this issue, maybe some other merged PR fixed this. Can you confirm?
If yes, please close this PR.

**Comment by prashantkh19** [comment]
If you still find this issue, please share a gif of the same.

**Comment by therajanmaurya** [comment]
It does not exists anymore.

**Pull request closed without merging**